src/ripple/app/misc/Transaction.h

```diff
@@ -166,6 +278,18 @@ class Transaction
     TER             mResult = temUNCERTAIN;
     bool            mApplying = false;
 
+    /** different ways for transaction to be accepted */
+    bool            mApplied = false;
+    bool            mQueued = false;
+    bool            mBroadcast = false;
+    bool            mKept = false;
+
+    /** tips on current state of the ledger */
+    LedgerIndex     mValidatedLedger = 0;
+    XRPAmount       mMinFeeRequired = 0;
+    std::uint32_t   mAccountSeqNext = 0;
+    std::uint32_t   mAccountSeqAvail = 0;
```

Comment:
Rather than using `0` as a `not set` value, consider using `boost::optional`

Reply:
What do you think about replacing these four member vars with a `boost::optional<CurrentLedgerState> mCurrentLedgetState;` where `CurrentLedgerState` just packages up these vars?

Reply:
Done.